internal/generate/clusterserviceversion/clusterserviceversion.go

```diff
@@ -78,6 +78,20 @@ type getBaseFunc func() (*operatorsv1alpha1.ClusterServiceVersion, error)
 // Option is a function that modifies a Generator.
 type Option func(*Generator) error
 
+// WithBaseCreator creates a Generator's base CSV to a kustomize-style base.
+func WithBaseCreator(cfg *config.Config, inputDir, apisDir string, ilvl projutil.InteractiveLevel) Option {
+	return func(g *Generator) error {
+		//g.getBase = g.makeKustomizeBaseGetter(inputDir, apisDir, ilvl)
+		basePath := filepath.Join(inputDir, "bases", makeCSVFileName(g.OperatorName))
+		if genutil.IsNotExist(basePath) {
+			basePath = ""
+		}
+
+		g.getBase = g.makeBaseCreator(cfg, basePath, apisDir, requiresInteraction(basePath, ilvl))
+		return nil
+	}
+}
+
 // WithBase sets a Generator's base CSV to a kustomize-style base.
 func WithBase(inputDir, apisDir string, ilvl projutil.InteractiveLevel) Option {
```

Comment:
Once you switch to using `base.GetBase()` in `kustomize manifests`, you can change this to

```go
func WithBase(inputDir string) Option {
	return func(g *Generator) error {
		basePath := filepath.Join(inputDir, "bases", makeCSVFileName(g.OperatorName))
		if inputDir != "" && genutil.IsNotExist(basePath) {
			basePath = ""
		}
		base := bases.ClusterServiceVersion{BasePath: basePath}
		if g.baseCSV, err = base.GetBase(); err != nil {
			return fmt.Errorf("error getting ClusterServiceVersion base: %v", err)
		}
		return nil
	}
}
```

so the caller has control over what base CSV is used. You can populate GVKs in the generator from the collector. This makes the generator more easily testable and lets the caller decide whether to use a base or just update a bare CSV.